src/app/panel/auth/LogIn.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+import { TextInput, Button } from '@tremor/react';
+import { APIService } from '@/services/api';
+import { useRouter } from 'next/navigation';
+
+const LogIn: React.FC = () => {
+  const [username, setUsername] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [isFormValid, setIsFormValid] = useState<boolean>(false);
+  const router = useRouter();
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    console.log(`Username: ${username}, Password: ${password}`);
+    /*
+    Send the form to server, if username or password is wrong, show an error message.
+    */
+    const response = await APIService.signIn(username, password);
+    if (!response.ok) {
+      console.log('Error: ', response);
+      return;
+    }
+    else {
+      router.push('/panel');
+    }
+  };
+
+  const validateForm = (): void => {
+    if (username.length < 1) {
+      setIsFormValid(false);
+    }
+    else if (password.length < 1) {
+      setIsFormValid(false);
+    }
+    setIsFormValid(true);
+  }
+
+  const onUsernameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.target.value = e.target.value.replace(/[^a-zA-Z0-9]/g, '');
```

Comment:
Do we really want to replace these characters? I think we should instead change the validateForm function to check for invalid characters as well.

Reply:
I think it's better to directly not let the user use them, @Utkub24 what do you think? We can also give a warning with 'only these letter can be used'

Reply:
Yes, that's what I am saying.

Reply:
Yeah, let's not replace them. Instead let the user type it in but just don't allow it to be registered. I felt like my was not working when I tried to type @